scipy/optimize/_optimize.py

```diff
@@ -1413,6 +1419,11 @@ def _minimize_bfgs(fun, x0, args=(), jac=None, callback=None,
     xrtol : float, default: 0
         Relative tolerance for `x`. Terminate successfully if step size is
         less than ``xk * xrtol`` where ``xk`` is the current parameter vector.
+    c1 : float, default: 1e-4
```

Comment:
Are there guidelines for what the values of c1 and c2 should be? e.g. non-negative, in a range ...?

Reply:
These are the typical values used in [Numerical Optimization from Nocedal (p.54)](https://link.springer.com/book/10.1007/978-0-387-40065-5) and in [L-BFGS paper](https://link.springer.com/article/10.1007/BF01589116). They also must be such that $0 < c2 < c1 < 1$ this should potentially be check before running the line-search.